commit 4422f8e6b3a0095d60541db8a0d5772c2235f180 (refs/changes/01/1001/1)
Author: Chris Burroughs <chris.burroughs@joyent.com>
Date:   2016-11-30T13:56:05-05:00 (2 years, 10 months ago)
    
    MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
    MMMMMMMMMMMM        MMMMMMMMMMMM
    MMMMMMMMMM            MMMMMMMMMM
    MMMMMMMMM              MMMMMMMMM
    MMMMMMMM                MMMMMMMM
    MMMMMMM                 MMMMMMMM
    MMMMMMM                  MMMMMMM
    MMMMMMM                  MMMMMMM
    MMMMMMM    MMM    MMM    MMMMMMM
    MMMMMMM   MMMMM   MMMM   MMMMMMM
    MMMMMMM   MMMMM   MMMM   MMMMMMM
    MMMMMMMM   MMMM M MMMM  MMMMMMMM
    MMVKMMMM        M        MMMMMMM
    MMMMMMMM       MMM      MMMMMMMM
    MMMMMMMMMMMM   MMM  MMMMMMMMMMMM
    MMMMMMMMMM MM       M  MMMMMMMMM
    MMMMMMMMMM  M M M M M MMMMMMMMMM
    MMMM  MMMMM MMMMMMMMM MMMMM   MM
    MMM    MMMM M MMMMM M MMMM    MM
    MMM    MMMM   M M M  MMMMM   MMM
    MMMM    MMMM         MMM      MM
    MMM       MMMM     MMMM       MM
    MMM         MMMMMMMM      M  MMM
    MMMM  MMM      MMM      MMMMMMMM
    MMMMMMMMMMM  MM       MMMMMMM  M
    MMM  MMMMMMM       MMMMMMMMM   M
    MM    MMM        MM            M
    MM            MMMM            MM
    MMM        MMMMMMMMMMMMM       M
    MM      MMMMMMMMMMMMMMMMMMM    M
    MMM   MMMMMMMMMMMMMMMMMMMMMM   M
    MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
    
    WARNING: THESE DO NOT WORK, A MYSTERY!

diff --git a/lib/do_fwrule/do_delete.js b/lib/do_fwrule/do_delete.js
index b4369d8..a7b0fdd 100644
--- a/lib/do_fwrule/do_delete.js
+++ b/lib/do_fwrule/do_delete.js
@@ -31,10 +31,11 @@ function do_delete(subcmd, opts, args, cb) {
         return;
     }
 
-    var cli = this.top;
+    var tritonapi = this.top.tritonapi;
     var ruleIds = args;
 
-    vasync.pipeline({funcs: [
+    vasync.pipeline({arg: {cli: this.top}, funcs: [
+        common.cliSetupTritonApi,
         function confirm(_, next) {
             if (opts.force) {
                 return next();
@@ -61,8 +62,8 @@ function do_delete(subcmd, opts, args, cb) {
             vasync.forEachParallel({
                 inputs: ruleIds,
                 func: function deleteOne(id, nextId) {
-                    cli.tritonapi.deleteFirewallRule({
-                            id: id
+                    tritonapi.deleteFirewallRule({
+                        id: id
                     }, function (err) {
                         if (err) {
                             nextId(err);
diff --git a/lib/do_image/do_delete.js b/lib/do_image/do_delete.js
index a11501b..484841e 100644
--- a/lib/do_image/do_delete.js
+++ b/lib/do_image/do_delete.js
@@ -26,7 +26,8 @@ function do_delete(subcmd, opts, args, cb) {
     }
     var ids = args;
 
-    vasync.pipeline({arg: {}, funcs: [
+    vasync.pipeline({arg: {cli: this.top}, funcs: [
+        common.cliSetupTritonApi,
         /*
          * Lookup images, if not given UUIDs: we'll need to do it anyway
          * for the DeleteImage call(s), and doing so explicitly here allows
